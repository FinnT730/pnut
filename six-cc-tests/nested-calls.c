int abc(int x) {
  return x + 1;
}

int def(int x) {
  return x * 2;
}

int counter;
counter = 10;

int decrement(int i) {
  i--;
  return i;
}

int r[2801];

int main() {
  abc(123);
  r[abc(123)] = 2000;
  printf("%d \n", r[abc(123)]);
  /* while (counter = decrement(counter)) { */
  while (counter--) {
    printf("Counter: %d\n", counter);
  }
  return abc(def(123));
}
