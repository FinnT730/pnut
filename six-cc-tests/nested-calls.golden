2000
Counter: 9
Counter: 8
Counter: 7
Counter: 6
Counter: 5
Counter: 4
Counter: 3
Counter: 2
Counter: 1
Counter: 0
43
247
30504
30628
6 124
